packages/replay/src/types/replayFrame.ts

```diff
@@ -0,0 +1,174 @@
+import type { Breadcrumb, FetchBreadcrumbData, XhrBreadcrumbData } from '@sentry/types';
+
+import type { AllEntryData } from './performance';
+import type { EventType } from './rrweb';
+
+interface BaseReplayFrame {
+  timestamp: number;
+  /**
+   * For compatibility reasons
+   */
+  type: string;
+  category: string;
+  data?: Record<string, any>;
+  message?: string;
+}
+
+interface BaseDomFrameData {
+  nodeId?: number;
+  node?: {
+    id: number;
+    tagName: string;
+    textContent: string;
+    attributes: Record<string, any>;
+  };
+}
+
+/* Breadcrumbs from Core SDK */
+interface ConsoleFrameData {
+  logger: string;
+  arguments?: unknown[];
+}
+interface ConsoleFrame extends BaseReplayFrame {
+  category: 'console';
+  level: Breadcrumb['level'];
+  message: string;
+  data: ConsoleFrameData;
+}
+
+type ClickFrameData = BaseDomFrameData;
+interface ClickFrame extends BaseReplayFrame {
+  category: 'ui.click';
+  message: string;
+  data: ClickFrameData;
+}
+
+interface FetchFrame extends BaseReplayFrame {
+  category: 'fetch';
+  type: 'http';
+  data: FetchBreadcrumbData;
+}
+
+interface InputFrame extends BaseReplayFrame {
+  category: 'ui.input';
+  message: string;
+}
+
+interface XhrFrame extends BaseReplayFrame {
+  category: 'xhr';
+  type: 'http';
+  data: XhrBreadcrumbData;
+}
+
+/* Breadcrumbs from Replay */
+interface MutationFrameData {
+  count: number;
+  limit: boolean;
+}
+interface MutationFrame extends BaseReplayFrame {
+  category: 'replay.mutations';
+  data: MutationFrameData;
+}
+
+interface KeyboardEventFrameData extends BaseDomFrameData {
+  metaKey: boolean;
+  shiftKey: boolean;
+  ctrlKey: boolean;
+  altKey: boolean;
+  key: string;
+}
+interface KeyboardEventFrame extends BaseReplayFrame {
+  category: 'ui.keyDown';
+  data: KeyboardEventFrameData;
+}
+
+interface BlurFrame extends BaseReplayFrame {
+  category: 'ui.blur';
+}
+
+interface FocusFrame extends BaseReplayFrame {
+  category: 'ui.focus';
+}
+
+interface SlowClickFrameData extends ClickFrameData {
+  url: string;
+  timeAfterClickFs: number;
+  endReason: string;
+}
+interface SlowClickFrame extends BaseReplayFrame {
+  category: 'ui.slowClickDetected';
+  data: SlowClickFrameData;
+}
+
+interface OptionFrame {
+  sessionSampleRate: number;
+  errorSampleRate: number;
+  useCompressionOption: boolean;
+  blockAllMedia: boolean;
+  maskAllText: boolean;
+  maskAllInputs: boolean;
+  useCompression: boolean;
+  networkDetailHasUrls: boolean;
+  networkCaptureBodies: boolean;
+  networkRequestHasHeaders: boolean;
+  networkResponseHasHeaders: boolean;
+}
+
+export type BreadcrumbFrame =
```

Comment:
cc @ryan953 I updated the name of this type from `CrumbFrame` -> `BreadcrumbFrame`